### PR TITLE
feat(ui): Build redesigned KVM storage cards.

### PR DESCRIPTION
**Pull request by Caleb-Ellis**
## Done

- Built redesigned KVM storage cards.
- Built KVMMeter component which just wraps Meter with some specific styling for total/allocated/free data. We should be able to use it in other parts of the KVM resources summary.
- Removed `key` prop from Meter data. It was only used because the component maps over the data array - all it needs to be is a unique string so it probably didn't need to be included in the prop.

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Go to a KVM details page and check that the storage cards match [the design](https://app.zeplin.io/project/5f1a24be5ee8af8723f4e7fa/screen/5f33dc5944a05220896e857a)

## Fixes

Fixes canonical-web-and-design/MAAS-squad#2072

## Screenshot
0 0 0 0_8400_MAAS_r_kvm_190 (12)

### CODE REVIEW

File: ui/src/app/kvm/components/KVMMeter/KVMMeter.tsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  total: number;
+  unit: string;
+};
+
+const KVMMeter = ({
+  allocated,
+  free,
+  segmented = false,
+  total,
+  unit = "",
```
Comment (huwshimi) [inline ui/src/app/kvm/components/KVMMeter/KVMMeter.tsx:18]: This has a default value, but in the type it isn't optional...

**Review by huwshimi — APPROVED** [review]
A couple of small things.

File: ui/src/app/base/components/Meter/Meter.tsx [hunk 2]
```diff
@@ -104,7 +109,7 @@ const Meter = ({
           data.map((datum, i) => (
             <div
               className="p-meter__filled"
-              key={`${datum.key}-bar`}
+              key={generateKey()}
```
[the remark this answers is not in the document]
Comment (Caleb-Ellis) [reply ui/src/app/base/components/Meter/Meter.tsx:112]: Yeah that's much simpler

File: ui/src/app/kvm/components/KVMMeter/KVMMeter.tsx (hunk 1 above)
Comment (Caleb-Ellis) [reply ui/src/app/kvm/components/KVMMeter/KVMMeter.tsx:18]: thanks, fixed!

**Pull request merged**